testsys/src/run_aws_ecs.rs

```diff
@@ -111,51 +144,199 @@ impl RunAwsEcs {
+                let upgrade_test_name = format!("{}-upgrade-migration-test", self.name);
+                let upgraded_test_name = format!("{}-upgraded-test", self.name);
+                let downgrade_test_name = format!("{}-downgrade-migration-test", self.name);
+                let final_test_name = format!("{}-final-test", self.name);
+                let init_ecs_test = self.ecs_test(
+                    &init_test_name,
+                    aws_secret_map.clone(),
+                    &ec2_resource_name,
+                    cluster_resource_name,
+                    None,
+                )?;
+                let upgrade_test = self.migration_test(
+                    &upgrade_test_name,
+                    upgrade_version,
+                    tuf_repo.clone(),
+                    aws_secret_map.clone(),
+                    &ec2_resource_name,
+                    cluster_resource_name,
+                    Some(vec![init_test_name.clone()]),
+                    migration_agent_image,
+                    &self.migration_agent_pull_secret,
+                )?;
+                let upgraded_ecs_test = self.ecs_test(
+                    &upgraded_test_name,
+                    aws_secret_map.clone(),
+                    &ec2_resource_name,
+                    cluster_resource_name,
+                    Some(vec![init_test_name.clone(), upgrade_test_name.clone()]),
+                )?;
+                let downgrade_test = self.migration_test(
+                    &downgrade_test_name,
+                    starting_version,
+                    tuf_repo,
+                    aws_secret_map.clone(),
+                    &ec2_resource_name,
+                    cluster_resource_name,
+                    Some(vec![
+                        init_test_name.clone(),
+                        upgrade_test_name.clone(),
+                        upgraded_test_name.clone(),
+                    ]),
+                    migration_agent_image,
+                    &self.migration_agent_pull_secret,
+                )?;
+                let final_ecs_test = self.ecs_test(
+                    &final_test_name,
+                    aws_secret_map.clone(),
+                    &ec2_resource_name,
+                    cluster_resource_name,
+                    Some(vec![
+                        init_test_name,
+                        upgrade_test_name,
+                        upgraded_test_name,
+                        downgrade_test_name,
+                    ]),
+                )?;
+                vec![
+                    init_ecs_test,
+                    upgrade_test,
+                    upgraded_ecs_test,
+                    downgrade_test,
+                    final_ecs_test,
+                ]
+            } else {
+                return Err(error::Error::InvalidArguments{why:"If performing an upgrade/downgrade test, `starting-version`, `upgrade-version` must be provided.".to_string()});
```

Comment:
I'm not sure why fmt is allowing this. Can you try manually breaking this up onto a couple of lines such that the long string ends up on its own line?